src/qml/DigitizingToolbar.qml

```diff
@@ -128,6 +151,33 @@ VisibilityFadingRow {
     }
   }
 
+  Dialog {
+    id: cancelDialog
+    parent: mainWindow.contentItem
+
+    visible: false
+    modal: true
+
+    x: ( mainWindow.width - width ) / 2
+    y: ( mainWindow.height - height ) / 2
+
+    title: qsTr( "Cancel digitizing" )
+    Label {
+      width: parent.width
+      wrapMode: Text.WordWrap
+      text: qsTr( "You are about to discard the digitized geometry, is that your intention?" )
```

Comment:
I'm always pro prose. But don't most people like it shorter? Like "Should the digitized geometry be discarded?" It's up to you.